proposals/NNNN-experiment-tracking/README.md

```diff
@@ -0,0 +1,495 @@
+
+The proposal tackles these issues by:
+
+- **Expanding Model Registry** into a central experiment tracking store for experiments, runs, metrics, and artifacts
+  across all Kubeflow components.
+- Providing **MLFlow SDK compatibility**, enabling users to leverage familiar APIs while storing data in Kubeflow.
+- Implementing **multitenancy** with Kubernetes namespace-level isolation.
+- Offering a **unified UI** for comprehensive experiment management and visualization.
+
+This approach will enable users to log experiments and compare runs from various sources in a single interface while
+maintaining consistent metadata throughout the ML lifecycle. This has the added benefit of eliminating technical debt
+associated with Kubeflow Pipelines' heavy reliance on MLMD.
+
+## Motivation
+
+Currently, experiment tracking in Kubeflow is fragmented and tightly coupled to specific components. Kubeflow Pipelines
+requires users to run everything through pipelines to track experiments, while Katib's experiments are limited to
+hyperparameter tuning scenarios. This fragmentation creates several challenges:
+
+1. **Limited flexibility**: Users cannot log experiments directly from Python scripts, Jupyter Notebooks, or other
+   Kubeflow components without forcing everything through pipelines. This restriction often drives users to seek
+   solutions outside the Kubeflow ecosystem.
+1. **Fragmented experience**: Users must navigate multiple interfaces to correlate run results and evaluation data,
+   depending on which Kubeflow component they use (e.g., Pipeline runs, Katib experiments, Training Operator results).
+1. **No unified tracking**: Users cannot easily compare runs, share insights, or maintain consistent metadata across the
+   entire Kubeflow ecosystem.
+1. **Maintenance challenges**: The Kubeflow Pipelines dependency on MLMD (Machine Learning Metadata) creates technical
+   debt and limits future flexibility due to its lack of active maintenance. MLMD also overlaps with the Model Registry
+   objectives defined in the
+   [Model Registry overview](https://www.kubeflow.org/docs/components/model-registry/overview/#use-case-2-experimenting-with-different-model-weights-to-optimize-model-accuracy),
+   creating separate metadata store experiences.
+
+These limitations make it difficult to effectively manage the machine learning lifecycle across Kubeflow.
+
+### Goals
+
+1. **Kubeflow centralized experiment tracking**: Create an experiment tracking system that is independent from Kubeflow
+   Pipelines, enabling experiments to be logged and managed without requiring pipeline execution.
+1. **Transform Model Registry into a unified metadata store**: Evolve the existing Model Registry component into a
+   centralized metadata store that can handle experiments, runs, metrics, and artifacts across the entire Kubeflow
+   ecosystem.
+1. **Pipelines integration**: Integrate Kubeflow Pipelines with the centralized experiment tracking system in Model
```

Comment:
rephrase emphasizing refactoring pipelines to externalize experiment tracking, removing the dependency (tech debt) from Pipelines. This allows pipelines to integrate with MR as well as 3rd party trackers.